lib/Config.js

```diff
@@ -385,6 +385,10 @@ function locationConstraintAssert(locationConstraints) {
             assert(locationConstraints[l].isCold === true);
         }
 
+        if (locationConstraints[l].type === 'crr') {
+            assert(locationConstraints[l].isCrr === true);
+        }
```

Comment:
If we always have the "type" crr and expect isCRR to be true, why not just relying on the `type` to detect if this is a CRR location instead?

Reply:
the point is actually the opposite: there may be multiple "types" of locations which support CRR (i.e. must be set in backbeat replication endpoint) but cannot be used for writing directly. So the flag `isCrr` was introduced to give this flexibility; and type may be used if we have different kinds of crr locations.